### ISSUE TITLE
Git pull does not delete files

### ISSUE THREAD

**Issue by bmiller** [3ll3d00d/beqdesigner#221]
The git pull operation for the beq files repo does not delete files when one is renamed.  A couple recent examples are John Wick 2, Underworld Blood Wars, or Ralph Breaks the Internet.  The files for all three were renamed during updates.  

It looks like this is an issue with porcelain's pull function from the research I've done so far.   I wanted to get it documented as a known issue here.  I'm looking for a solution to it since it's affecting the browser app too.

**Comment by 3ll3d00d** [comment]
looks like https://github.com/dulwich/dulwich/issues/452

**Comment by bmiller** [comment]
Yeah, I saw that bug the other night when I was looking into it.  I forgot to reference it.

**Comment by 3ll3d00d** [comment]
https://github.com/dulwich/dulwich/issues/588 has a possible workaround for something that looks a bit more like a working `git reset --hard`

**Comment by JanSam** [comment]
As discussed in the avs thread lately, a button to purge the BEQ files(under preferences) would be great too, not just for the merge portion of the BEQd.

**Comment by bmiller** [comment]
@3ll3d00d have you looked into pygit2 before?  It uses libgit2 under the hood which complicates things a little for building.  I was going to use it with a Swift library for the iOS app I'm working on and thought of checking on it for BEQd.  I could port my iOS stuff to python after I do it if there wasn't going to be issues doing so.

**Comment by 3ll3d00d** [comment]
a quick search suggests it has it's own problems as well as bringing packaging complexity inc the need to build from source on MacOS (e.g. https://github.com/iterative/dvc/issues/2215),
